core/trino-main/src/main/java/io/trino/operator/TaskStats.java

```diff
@@ -87,6 +88,10 @@ public class TaskStats
     private final DataSize physicalWrittenDataSize;
     private final Optional<Integer> maxWriterCount;
 
+    // scale writers metrics
```

Comment:
so this is part of task stats, but it doesn't seem to be propagated to stage stats. Therefore one has to go really deep into query/stage json to get to the details. Do you think we could somehow aggregate the stats? I think distribution `partitionsAssigned` should be fine to aggregate. I'm not sure about `long scalingCount, double scaledPartitionPercentage`. Maybe there is a way to express these in metrics that can be aggregated (e.g. some kind of distribution or weighted average)